archetype/maven-url-handler/src/main/java/io/helidon/build/archetype/maven/url/mvn/MavenResolver.java

```diff
@@ -45,16 +46,24 @@ File resolve(String url) throws IOException {
         }
         url = url.substring((PROTOCOL + "://").length());
         Parser parser = new Parser(url);
-        return resolve(parser.getArchivePath());
+
+        return resolve(parser, parser.getType());
     }
 
-    private File resolve(String[] path) throws IOException {
+    private File resolve(Parser parser, String type) throws IOException {
+        String fileName = parser.getArtifactId() + "-" + parser.getVersion() + "." + parser.getType();
         Visitor visitor = new Visitor(getLocalRepository());
-        return visitor.visit(path);
+        visitor.visit(parser.getArchivePath());
+        return type.equals("jar")
+                ? visitor.visitJar(fileName, parser.getPathFromArchive())
+                : visitor.visitZip(fileName, parser.getPathFromArchive());
     }
 
     private File getLocalRepository() {
-        return new File(System.getProperty("user.home"), ".m2/repository");
+        String local = System.getProperty(SYSTEM_PROPERTY_LOCAL_REPO);
+        return local == null
```

Comment:
Let's not hard-code any default here, instead throw an exception and make it fail.